### PR TITLE
INT: add support for creating methods in CreateFunctionIntention

### PR DESCRIPTION
**Pull request by Kobzol**
This PR adds support for creatng methods in `CreateFunctionIntention`. Also I previously forget to register the intention and add a description, since it was originally a quick fix, so this PR also fixes that.

create-method

### CODE REVIEW

**Comment by vlad20012** [comment]
> Also I previously forget to register the intention and add a description, since it was originally a quick fix, so this PR also fixes that.

😄 we must add some check to `RsIntentionTestBase`

File: src/main/kotlin/org/rust/ide/intentions/CreateFunctionIntention.kt [hunk 1]
```diff
@@ -15,123 +15,198 @@ import org.rust.ide.presentation.renderInsertionSafe
...
+            if (!functionCall.expr.isAncestorOf(path)) return null
+
+            val module = getTargetModuleForFunction(path) ?: return null
+            val name = path.referenceName
+            text = "Create function `$name`"
+            return Context.Function(functionCall, name, module)
+        }
+        val methodCall = element.parentOfType<RsMethodCall>()
+        if (methodCall != null) {
+            if (methodCall.reference.resolve() != null) return null
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/ide/intentions/CreateFunctionIntention.kt:69]: Same as in function call case, let's limit possible caret position to the method identifier. `element == methodCall.identifier` should works

**Comment by Kobzol** [comment]
>  we must add some check to `RsIntentionTestBase`

Ooh, I actually thought that it exists, I even wrote so on my blog :D But it only exists for inspections. I'll add it for intentions too.

**Review by Undin — CHANGES_REQUESTED** [review]
@Kobzol Let's split this PR into two: the first to fix intention registration and the second one to add new feature. It allows us to cherry-pick only fix into v130 release, put new feature into v131 (or later if something will happen) and don't change our usual workflow

**Review by vlad20012 — APPROVED** [review]
:+1: 
Rebase it to the latest master, please

**Pull request merged**